integration_tests/stack_reference/step1/Program.cs

```diff
@@ -14,8 +15,8 @@ static Task<int> Main(string[] args)
             var slug = $"{Deployment.Instance.OrganizationName}/{Deployment.Instance.ProjectName}/{Deployment.Instance.StackName}";
             var a = new StackReference(slug);
 
-            var oldVal = (string[])await a.GetValueAsync("val");
-            if (oldVal.Length != 2 || oldVal[0] != "a" || oldVal[1] != "b")
+            var oldVal = (ImmutableArray<object>)await a.GetValueAsync("val");
+            if (oldVal.Length != 2 || (string)oldVal[0] != "a" || (string)oldVal[1] != "b")
```

Comment:
Concerning that I had to make these changes. Was this test code not running before?

I haven't investigated further, but maybe the older version of ProgramTest wasn't erroring when it couldn't find the `step1` and `step2` dirs and was just running the original Program.cs? It is erroring now with the `pulumi/pulumi` upgrade in go.mod (`step1` dir doesn't exist), which led me to the `filepath.Join` changes in [integration_tests/integration_dotnet_smoke_test.go](https://github.com/pulumi/pulumi-dotnet/pull/279/files#diff-e482defbbadda45dbe4697cd92fc71e3d787506040e7b63ec13355c35750ffba).

But with that fixed, now running into this problem in the test code itself. The actual returned value from `a.GetValueAsync` is an `ImmutableArray<object>` not `string[]`. Possible regression?

Reply:
I don't think a regression, this test just never ran before.